internal/pkg/cli/pipeline_show.go

```diff
@@ -87,34 +86,63 @@ func newShowPipelineOpts(vars showPipelineVars) (*showPipelineOpts, error) {
 	return opts, nil
 }
 
-// Validate returns an error if the flag values passed by the user are invalid.
+// Validate returns an error if the optional flag values passed by the user are invalid.
 func (o *showPipelineOpts) Validate() error {
 	if o.appName != "" {
 		if _, err := o.store.GetApplication(o.appName); err != nil {
-			return err
+			return fmt.Errorf("validate application name: %w", err)
```

Comment:
Interesting! Just to call this out - consider `copilot pipeline show --app non-existing-app --name my-pipeline`. We only need `appName` when we need to ask for pipeline name. So in this case,  the `--app` flag isn't really needed. Then the question is, should we still validate app?

The way it is here is to still validate the app. I think this is a nice way to sort of "remind" user that they may be thinking about a wrong app / wrong pipeline and I agree with the current approach. 

While for the other commands we do `validateOrAskAppName` in `Ask` instead of splitting the validation and ask, splitting them this way actually makes more sense for `pipeline show` because of the special nature of `--app` flag here.